File: src/main/java/de/stea1th/greetings/entity/Person.java
package de.stea1th.greetings.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@EqualsAndHashCode(callSuper = true)
@Entity
@Data
@NoArgsConstructor
public class Person extends AbstractBaseEntity{

    @NotBlank
    private String firstName;

    @NotBlank
    private String lastName;

    @NotNull
    private LocalDate dateOfBirth;

    @NotBlank
    @Email
    private String email;

    @NotNull
    private boolean autoEmailGreetings = false;

}

File: src/main/webapp/js/person.js
$(function () {

    getAll();

});

function getPersonName(id) {
    var url = "/api/greetings/person?id=" + id;
    $.get(url).done(function (data) {
        $('#person-name').text(data.firstName);
    })
}

function getAll() {
    var url = "/api/greetings/";
    $.get(url).done(function (data) {
        for (var i = 0; i < data.length; i++) {
            var person = data[i];
            var row = "<tr>" +
                // "<th>" + person.id + "</th>" +
                "<th>" + person.firstName + "</th>" +
                "<th>" + person.lastName + "</th>" +
                "<th>" + person.dateOfBirth + "</th>" +
                "<th>" + person.email + "</th>" +
                "<th>" + person.autoEmailGreetings + "</th>" +
                "</tr>";
            $("tbody").append(row);
        }
    });

}
